### PR TITLE
feat: org sign-up UI

### LINKED ISSUE #2822: Sign-up UI updates

### PR DESCRIPTION
**Pull request by BruceMacD**
## Summary
Add orgs to Infra sign-up UI (note that it is going into the org_collab branch, not main).

- Updates UI
- Updates sign-up for separate org and domain
- Does not require org name to be unique

image

image

### Next steps 
#2801 
- Update the quickstart to direct new self-hosted deployments to the "sign-up" UI
- Update ingress when a new org subdomain is created
- https://github.com/infrahq/infra/issues/2806

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [ ] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #2822

### CODE REVIEW

File: internal/server/models/organization.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 type Organization struct {
 	Model
 
-	Name      string `gorm:"uniqueIndex:idx_organizations_name,where:deleted_at is NULL"`
+	Name      string
```
Comment (BruceMacD) [inline internal/server/models/organization.go:14]: I removed the need for this to be a unique index. We can uniquely identify orgs by their domain. They can name it as they like.

File: ui/pages/signup/index.js [hunk 2]
```diff
@@ -148,8 +165,64 @@ export default function Signup() {
...
+            }`}
+          />
+          {errors.orgName && <ErrorMessage message={errors.orgName} />}
+        </div>
+        <div className='my-2 w-full'>
+          <label
+            htmlFor='orgDoman'
+            className='text-3xs uppercase text-gray-500'
+          >
+            Domain
```
Comment (BruceMacD) [inline ui/pages/signup/index.js:198]: I'd like to put a tooltip here that explains what the domain is for, but that can be a separate change.

File: ui/pages/index.js [hunk 3]
```diff
@@ -1,7 +1,14 @@
 import { useRouter } from 'next/router'
+import { useEffect } from 'react'
 
 export default function Index() {
-  useRouter().replace('/destinations')
+  const router = useRouter()
```
Comment (BruceMacD) [inline ui/pages/index.js:5]: Removing the sign-up API check exposed this issue where the router could be called during server-side rendering. Putting the re-route in the use effect ensure this is only called client-side.

File: ui/pages/signup/index.js [hunk 4]
```diff
@@ -54,10 +54,15 @@ export default function Signup() {
         throw await res.json()
       }
 
-      await mutate('/api/signup')
-      await mutate('/api/users/self')
-
-      router.replace('/')
+      let host = window.location.host
+      host = host.replace('www.', '')
+      location =
```
Comment (jmorganca) [inline ui/pages/signup/index.js:59]: Use a template string here

```
window.location = `${window.location.protocol}//${created.org}.${host}/login`
```

File: ui/__test__/signup.test.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
```
Comment (hoyyeva) [inline ui/__test__/signup.test.js:1]: ❤️ ❤️‍🔥

**Review by hoyyeva — APPROVED** [review]

**Comment by BruceMacD** [comment]
Closing this because the tests won't run, will re-open

**Pull request closed without merging**